src/source/shift.rs

```diff
@@ -4,141 +4,136 @@ use json;
 use json::object::Object;
```

Comment:
While the changes that you're making are a great improvement, I imagine that in some cases, the additional escaping cost will be non-trivial. Perhaps we should document this somewhere?

Reply:
What do you mean by the additional escaping cost? The only case I can think of where something new is happening is when passing strings to the codegen.

Reply:
There's one more level of string escaping as there used to be, no? If not, ignore my comment :)

Reply:
No, in most places escaping is the same as before. It always used to call `encodeWTF8` on top of `JSON.stringify` just like now, except it was all hidden inside a JS wrapper whereas now this is done in separate modules.